### PR TITLE
Added setPosition functions to SpartronicsMax and SpartronicsSRX

### PR DESCRIPTION
**Pull request by thehappycamden**

### CODE REVIEW

File: src/main/java/com/spartronics4915/lib/hardware/motors/SpartronicsEncoder.java [hunk 1]
```diff
@@ -20,15 +20,20 @@ public double getVelocity() {
         public double getPosition() {
             return 0;
         }
+
+        @Override
+        public void setPosition(double position) {
+            Logger.error("Couldn't set the phase of a disconnected encoder");
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/lib/hardware/motors/SpartronicsEncoder.java]: This logger message doesn't make sense.

**Review by pietroglyph — CHANGES_REQUESTED** [review]

**Comment by thehappycamden** [comment]
Also added Indexer code, pull request hadn't been accepted yet.

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 2]
```diff
@@ -1,65 +1,170 @@
...
+        mOpticalFlag = new DigitalInput(Constants.Indexer.kOpticalFlagId);
+
+        // Setup Prox Sensor for indexing
+        mProxSensor = new DigitalInput(Constants.Indexer.kProxSensorId);
     }
 
     // Outline your API here by creating specific methods.
     // Each method should perform a _singular action_
     // - eg. instead of a setIntake method, control each intake motor individually
     // setIntake functionality should be implemented in a command.
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Remove this

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 3]
```diff
@@ -1,65 +1,170 @@
...
+        double deltaPosition = 0.25 * ((double) N); // Cast N to double and convert to rotations
+        targetPosition += deltaPosition;
+        mSpinner.setPosition(targetPosition); // Rotate Spinner to target.
     }
 
-    public int rotateEmpty()
+    /**
+     * Returns spinner to "0" position on the encoder
+     */
+    public void returnSpin()
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Change to `returnToHome`?

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 4]
```diff
@@ -1,65 +1,170 @@
...
-        return 0;
+        targetPosition = 0;
+        mSpinner.setPosition(targetPosition);
     }
 
-    public void loadBall()
+    /**
+     * Move spinner to nearest position
+     */
+    public void endSpinner()
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Change to `toNearestQuarterRotation`?

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 5]
```diff
@@ -1,65 +1,170 @@
...
     {
+        mIsLaunching = false;
+        mLoader.setVelocity(0);
+    }
 
+    /**
+     * 
+     * @return whether or not the loader motor is running
+     */
+    public boolean getLaunching()
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Change to `isLaunching`?

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 6]
```diff
@@ -1,65 +1,170 @@
...
     {
+        // Rotates to nearest quarter rotation
+        targetPosition = Math.ceil(mSpinner.getEncoder().getPosition() * 4) / 4;
+        mSpinner.setPosition(targetPosition);
+    }
 
+    /**
+     * Start loading balls into the shooter
+     */
+    public void load()
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Change to `loadBalls`?

**Review by pietroglyph — CHANGES_REQUESTED** [review]
Just a couple of naming nitpicks.

**Comment by thehappycamden** [comment]
yay

**Pull request merged**